### PR TITLE
Add colorized console logs to Appium (1.x)

### PR DESCRIPTION
**Pull request by mikesalvia**
## Proposed changes

Add colorized console logs to Appium (1.x)

## Types of changes

What types of changes does your code introduce to Appium?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [X] I have read the [CONTRIBUTING](https://github.com/appium/appium/blob/master/CONTRIBUTING.md) doc
- [X] I have signed the [CLA](https://cla.js.foundation/appium/appium)
- [X] Lint and unit tests pass locally with my changes
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added necessary documentation (if appropriate)
- [X] Any dependent changes have been merged and published in downstream modules

## Screenshot
Screen Shot 2022-09-12 at 3 45 01 PM

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -62,11 +62,13 @@
     "asyncbox": "2.x",
     "axios": "^0.x",
     "bluebird": "3.x",
+    "chalk": "4.1.2",
     "continuation-local-storage": "3.x",
     "find-root": "^1.1.0",
     "lodash": "^4.17.11",
     "longjohn": "^0.2.12",
     "npmlog": "4.x",
+    "rainbow-colors-array": "^0.0.31",
```
Comment (jlipps) [inline package.json:71]: i would probably not pull in this dep. we could get away with a short list of colours in our source code. besides, this doesn't seem to be an open source project, or at least i couldn't find a github link

File: lib/logsink.js [hunk 2]
```diff
@@ -183,6 +183,23 @@ async function createTransports (args) {
...
+  if (prefixesEncountered.indexOf(prefixId) > -1) {
+    return '#' + colorList[prefixesEncountered.indexOf(prefixId)].hex;
+  }
+  prefixesEncountered.push(prefixId);
+  const color = colorList[nextColorIndex];
+  nextColorIndex++;
+  return '#' + color.hex;
+}
+
+import rca from 'rainbow-colors-array';
```
Comment (jlipps) [inline lib/logsink.js:198]: all imports go up top

File: package.json (hunk 1 above)
Comment (mikesalvia) [reply package.json:71]: Yeah, I thought we might be able to pull this out. I can actually just generate it and copy it as a JSON array.

Comment (mikesalvia) [reply package.json]: @jlipps Fixed!

File: lib/logsink.js (hunk 2 above)
Comment (mikesalvia) [reply lib/logsink.js]: @jlipps Fixed!

**Comment by boneskull** [comment]
@jlipps This targets Appium 1.x.  I thought we were no longer developing that?

**Comment by mikesalvia** [comment]
@boneskull This originated as an update to 1.22.3 for Appium Desktop, but it made sense to bring it over to Appium, and I can easily port this to 2.0 if needed.

**Comment by jlipps** [comment]
@mikesalvia ah i didn't catch that it targeted 1.x. we won't be making any more 1.x releases, so this code would never go anywhere (or get into appium desktop), unless you build it yourself. it would probably be better to target 2.0 with it.

and yes, unfortunately it means the change won't get into appium desktop unless/until we update it to run appium 2

**Comment by mikesalvia** [comment]
@jlipps OK let's close this one up.

**Pull request closed without merging**